### PR TITLE
Electron4 changes

### LINKED ISSUE #289: Upgrade Electron.
Current version is behind. Electron is at 2.X.

-----

We could/should upgrade other deps as well in this PR.

### PR DESCRIPTION
**Pull request by joshuef**
Fixes #289 

Also removes react-icons #492 #455 

QA: 
- [ ] Regression checks.
- [ ] Check add/remove buttons exist (they will have new style from antd).

### CODE REVIEW

File: app/extensions/safe/ffi/authenticator.ts [hunk 1]
```diff
@@ -42,8 +42,8 @@ const _decodeReqPool = Symbol( 'decodeReqPool' );
  * characters or symbols which are not valid for a URL like '=' sign,
  * and making it lower case.
  */
-const genAppUri = str => {
-    const urlSafeBase64 = new Buffer( str )
+const genAppUri = ( str ) => {
+    const urlSafeBase64 = new Buffer.from( str )
```
Comment (hunterlester) [inline app/extensions/safe/ffi/authenticator.ts]: I'm surprised that no warnings were thrown for this syntax. Shouldn't it either be `Buffer.from` or `new Buffer`?

File: app/extensions/safe/ffi/refs/parsers.ts [hunk 2]
```diff
@@ -93,14 +93,14 @@ export const parseContainerReq = containersReq => ( {
     containers_cap: containersReq.containers_cap
 } );
 
-const parseXorName = str => {
-    const b = new Buffer( str );
+const parseXorName = ( str ) => {
+    const b = new Buffer.from( str );
```
Comment (hunterlester) [inline app/extensions/safe/ffi/refs/parsers.ts]: `Buffer.from`

File: app/extensions/safe/ffi/refs/parsers.ts [hunk 3]
```diff
@@ -93,14 +93,14 @@ export const parseContainerReq = containersReq => ( {
...
     if ( b.length !== 32 ) throw Error( 'XOR Names _must be_ 32 bytes long.' );
     const name = types.XorName( b );
-    return new Buffer( name ).toString( 'hex' );
+    return new Buffer.from( name ).toString( 'hex' );
```
Comment (hunterlester) [inline app/extensions/safe/ffi/refs/parsers.ts]: `Buffer.from`

File: app/extensions/safe/ffi/refs/parsers.ts [hunk 4]
```diff
@@ -137,9 +137,9 @@ export const parseUserMetaDataArray = ( metaArr, len ) => {
     return res;
 };
 
-const parseAppAccessInfo = appAccess => {
-    let signKey = types.U8Array( new Buffer( appAccess.sign_key ) );
-    signKey = new Buffer( signKey ).toString( 'hex' );
+const parseAppAccessInfo = ( appAccess ) => {
+    let signKey = types.U8Array( new Buffer.from( appAccess.sign_key ) );
```
Comment (hunterlester) [inline app/extensions/safe/ffi/refs/parsers.ts]: `Buffer.from`

File: app/extensions/safe/ffi/refs/parsers.ts [hunk 5]
```diff
@@ -137,9 +137,9 @@ export const parseUserMetaDataArray = ( metaArr, len ) => {
...
+    signKey = new Buffer.from( signKey ).toString( 'hex' );
```
Comment (hunterlester) [inline app/extensions/safe/ffi/refs/parsers.ts]: `Buffer.from`

**Comment by hunterlester** [comment]
Looking into why notifications are not showing

File: app/extensions/safe/ffi/authenticator.ts (hunk 1 above)
Comment (joshuef) [reply app/extensions/safe/ffi/authenticator.ts]: Ah yeh, probably should be. I'll update this.

**Comment by joshuef** [comment]
@hunterlester FYI buffer stuff removed from this PR as was wreaking unexpected havoc w/ CI

**Comment by hunterlester** [comment]
@joshuef Everything appears to be working nicely, except for the `did-fail-load` event still not firing on `webview` when a web request is blocked.
I'm using this branch and your `Bviews` branch to rebase my `651`, BrowserView draft, where a blocked web request is firing a `did-fail-load` on BrowserView `webContents`.

**Comment by joshuef** [comment]
@hunterlester just to be clear, this is working in master, correct?

**Comment by hunterlester** [comment]
@joshuef Correct, working in master.

**Comment by joshuef** [comment]
Hmmmm ... I'm unable to find if triggering `did-fail-load` is something that is 'supposed' to happen with blocked reqs in electron.

Which raises a Q:

Do we simply need to trigger the notification elsewhere for this PR to be valid?

Or do we shelve electron 4 upgrades until browserview, which behaves as master it sounds like (and also implies that maybe `did-fail-load` should be triggered)

**Comment by joshuef** [comment]
Also: i'm noticing that the shortcut to select the address bar is not working atm.

**Pull request closed without merging**